src/evm.c

```diff
@@ -283,6 +287,7 @@ static inline bool ensureMemory(context_t *callContext, uint64_t capacity) {
         }
         callContext->gas -= memoryGas;
     }
+    memory_ensure(&callContext->memory, capacity);
```

Comment:
because this accesses `num_uint8s`, the undefined behavior likely originates from this change.

Reply:
fixed in 722bc41